operator/duties/controller.go

```diff
@@ -0,0 +1,164 @@
+package duties
+
+import (
+	"context"
+	"encoding/hex"
+	"github.com/bloxapp/eth2-key-manager/core"
+	"github.com/bloxapp/ssv/beacon"
+	"github.com/bloxapp/ssv/validator"
+	"github.com/herumi/bls-eth-go-binary/bls"
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/shared/slotutil"
+	"go.uber.org/zap"
+	"time"
+)
+
+// dutyExecutor represents the component that executes duties
+type dutyExecutor interface {
+	ExecuteDuty(duty *beacon.Duty) error
+}
+
+// DutyController interface for dispatching duties execution according to slot ticker
+type DutyController interface {
+	Start()
+}
+
+// ControllerOptions holds the needed dependencies
+type ControllerOptions struct {
+	Logger              *zap.Logger
+	Ctx                 context.Context
+	BeaconClient        beacon.Beacon
+	EthNetwork          core.Network
+	ValidatorController validator.IController
+	GenesisEpoch        uint64
+	DutyLimit           uint64
+}
+
+// dutyController internal implementation of DutyController
+type dutyController struct {
+	logger              *zap.Logger
+	ctx                 context.Context
+	ethNetwork          core.Network
+	executor            dutyExecutor
+	fetcher             DutyFetcher
+	validatorController validator.IController
+	genesisEpoch        uint64
+	dutyLimit           uint64
+}
+
+var secPerSlot int64 = 12
```

Comment:
@amir-blox I'm pretty sure we can get it from somewhere else no?
For the very least it shouldn't be here but rather in the beacon chain package

Reply:
We can use the entire implementation [prysm/shared/slotutil](https://github.com/prysmaticlabs/prysm/blob/develop/shared/slotutil/slottime.go#L21) instead of `getCurrentSlot`.

Note that currently I tweak that variable (`secPerSlot`) during tests to reduce test time, if we won't be able to do it